src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -197,81 +194,79 @@ class DataModelInstanceRelation(config: RelationConfig, modelExternalId: String)
           (index, field.name, propertyType)
       }
 
-    def parseRow(row: Row): DataModelInstance = {
+    def parseRow(row: Row): DataModelInstanceCreate = {
```

Comment:
If we curried it like `parseRow(row: Row)(schema: StructType)`, we could avoid the function in the function. `parseRow` now uses parameters that are in the outer function `fromRow`, and I think it's nicer to be able to have it all in one place!

Reply:
The beauty of this version is that the part in `fromRow` is done once and parseRow is called for each row. I'll try moving the first part out to another function.

Reply:
That _is_ nice.. but I think we get the same benefit from currying.. Do you know @wjoel ?

Reply:
I thought it is just a way of grouping the arguments? 🤔

Reply:
I may not be using the term right, but essentially having multiple argument lists to the function, so that we can do
`def parseRow(row: Row)(schema: StructType): DataModelInstanceCreate`
and then use it like
`parseRow(someSchema)`, which will then return a function of the type `Row => DataModelInstanceCreate`..

Seems the actual definition is https://www.baeldung.com/scala/currying#currying

> Currying is the process of converting a function with multiple arguments into a sequence of functions that take one argument. Each function returns another function that consumes the following argument.

Reply:
I thought you wanted me to remove the inner function and make it one, sorry I misunderstood

Reply:
we don't make use of `schema: StructType` directly in `parseRow`. It uses an array calculated in `fromRow` using the schema. Should I take that one as input?